QCircLearning/trainerModel.py

```diff
@@ -130,8 +132,11 @@ def NN_opt(func, x0, callback=None, **kwargs):
                         f"Run ID: {kwargs['run_id']}, Epoch {epoch + 1}/{classical_epochs}, Average Loss: {total_loss:.1e}"
                     )
                     sys.stdout.flush()
+                if early_stopping.counter == 0:
+                    best_model = copy.deepcopy(model)
```

Comment:
To clarify the logic here:
```suggestion
                best_model = copy.deepcopy(model) if early_stopping.reset else best_model
```

And define the attribute `reset` in `EarlyStopping`:

```python
@property
def reset(self):
    if self.counter == 0
```

Also, we should move this to the beginning of the loop. Since we will call `model.eval()` regardless, it doesn't make sense to assign `None` to the `best_model`

Reply:
Or maybe just keep it at the bottom to confirm the update. But, since we will assign the model at least once, we don't have to define `best_model` outside the loop.